### PR TITLE
Redraw longterm_change_in_programming_usage as slopegraph (fix #215)

### LINKED ISSUE #215: Considered slope charts to compare pre- & post-workshop data or other time series?
Hello :-)

I would like to try converting one or the other plot into a [slope chart/graph](http://charliepark.org/slopegraphs/). For example: https://github.com/carpentries/assessment/blame/master/learner-assessment/reports-src/2018-07-learners-pre-post-surveys.Rmd#L1066-L1103

Do you have any comments about whether or not a slope chart/graph would be useful or feasible in the reports? Any objections against such a PR?

If not, please feel free to assign the issue to me. I'm starting a [local study group](https://github.com/TIBHannover/2018-07-09-FAIR-Data-and-Software/pull/12), for which that may be a neat datavis example.

Thanks!

### PR DESCRIPTION
**Pull request by katrinleinweber**
It's based on https://github.com/jkeirstead/r-slopegraph/pull/1 and looks like this:

grafik

I tried removing some of the whitespace between the labels on the left and the data, but I haven't found which part of [`theme_slopegrap()`](https://github.com/jkeirstead/r-slopegraph/blob/master/slopegraph.r#L166-L187) is responsible. Maybe the whitespace adjustment occurs somewhere else?

---

There is also @leeper's [`slopegraph` package]( https://github.com/leeper/slopegraph), but 

```
slopegraph::ggslopegraph2(
  dataframe = pre_and_post_usage,
  times = key,
  measurement = perc,
  grouping = value
  )
```

draws the doubly labelled:

grafik

I think that's due to the labels not being factors or levels or something, because sometime in between, I had them deduplicated and each label was only once on the far left and the far right :-/

---

So, to summarise, this is what slopegraphs would generally look like :-)

### CODE REVIEW

**Comment by kariljordan** [comment]
@tracykteal check out the slope graphs @katrinleinweber made for programming usage! What do you think? I'm going to merge this PR, but wanted you to see it first.

This is really awesome @katrinleinweber!

**Comment by tracykteal** [comment]
These are fantastic! A great visualization of this data. Thanks so much @katrinleinweber!

**Comment by katrinleinweber** [comment]
Thank you :-) Please let me rename `ProgrammingBefore` and `ProgrammingSince` to align with `Pre` and `Post` from the title. Also, upvotes to https://github.com/jkeirstead/r-slopegraph/pull/1 may help improve the `slopegraph`ing code generally ;-)

**Comment by katrinleinweber** [comment]
Ah, I got @leeper's way of drawing them to work:

grafik

The label sizes are a bit smaller and can't really be increased without overlapping with lines and text labels. [`slopegraph.R`](https://github.com/katrinleinweber/r-slopegraph/blob/536edc16aa6560979039b8e16d00323db1c740dd/slopegraph.r#L230) seems to avoid this with larger `geom_point(colour = "white", size = ...)`.

Which one do you like better?

**Comment by kariljordan** [comment]
I got an error in Line 26 when I tried to knit it:

`Error in open.connection(con, "rb") : HTTP error 404. Calls: <Anonymous> ... read_delimited -> read_connection -> open.connection Execution halted`

Do you know what this means and/or how I can fix it?

**Comment by katrinleinweber** [comment]
Oh right! I should have mentioned this, sorry. Just an incorrect filename but I updated it only in the other PR 217. Until that is merged, you can temporarily insert the missing `20` between ...`data/` and `180711`... ;-)

**Comment by katrinleinweber** [comment]
Note for next time: I'll push the knit-ed version after reviewing the source and before flagging a PR to be ready to be merged. Unless https://github.com/carpentries/lesson-infrastructure/issues/20 will be rolled out to this repo before ;-)

**Comment by kariljordan** [comment]
I tried installing slopegraph, but got this error:

`package ‘slopegraph’ is not available (for R version 3.4.1)`

I looked around online to get help and found that I should:
`install.packages("CGPfunctions")`

I'm pretty sure that's nor right. Any ideas?

**Comment by katrinleinweber** [comment]
Yes, it's not on CRAN (yet), see [leeper/slopegraph#installation](https://github.com/leeper/slopegraph#installation). `devtools::install_github("leeper/slopegraph")` works as well.

**Comment by kariljordan** [comment]
All [fixed](https://github.com/carpentries/assessment/commit/883936b6b26155d645c0c4c4c07bc8441e922015)!
